File: src/structural/bridge/python/classes/remote_controls/remote_control_with_volume.py
from classes.remote_controls.remote_control import RemoteControl


class RemoteControlWithVolume(RemoteControl):
    def volume_up(self) -> None:
        self.device.set_volume(self.device.get_volume() + 10)

    def volume_down(self) -> None:
        self.device.set_volume(self.device.get_volume() - 10)

File: src/structural/bridge/typescript/classes/RemoteControlWithVolume.ts
import { RemoteControl } from './RemoteControl';

export class RemoteControlWithVolume extends RemoteControl {
  volumeUp(): void {
    const volume = this.device.volume;
    this.device.volume = volume >= 100 ? volume : volume + 10;
  }
  volumeDown(): void {
    const volume = this.device.volume;
    this.device.volume = volume <= 0 ? volume : volume - 10;
  }
}
